scripts/sandbox.js

```diff
@@ -1,35 +1,67 @@
 let list = document.querySelector(".challenge-list");
 
 function addChallengesToDom(challenge) {
-  list.innerHTML += `
-     <li class="challenge-item">
-       <img class="challenge-image" alt="Hacker" src=${challenge.image} />
-       <h3 class="challenge-title">${challenge.title} (${challenge.type})</h3>
-       <ul
-       id="challenge-${challenge.id}
-         role="meter"
-         class="rating"
-         aria-label=${challenge.labels}
-         arial-valuemin="0"
-         aria-valuemax="5"
-         aria-valuenow=${challenge.rating}
-         aria-valuetext= " ${challenge.rating} out of 5"
-         
-       >
-         <li class="rating-star active"></li>
-         <li class="rating-star active"></li>
-         <li class="rating-star active"></li>
-         <li class="rating-star active"></li>
-         <li class="rating-star"></li>
-       </ul>
-       <small class="challenge-meta">${challenge.minParticipants}- ${challenge.maxParticipants} participants</small>
-       <p class="challenge-description">
-        ${challenge.description}
-       </p>
-       <button class="button primary modal-open">Book this room</button>
-     </li>
+  if (challenge.type === "onsite") {
+    list.innerHTML += `
+    <li class="challenge-item"> 
+      <img class="challenge-image" alt="Hacker" src=${challenge.image} />
+      <h3 class="challenge-title">${challenge.title} (${challenge.type})</h3>
+      <ul
+      id="challenge-${challenge.id}
+        role="meter"
+        class="rating"
+        aria-label=${challenge.labels}
+        arial-valuemin="0"
+        aria-valuemax="5"
+        aria-valuenow=${challenge.rating}
+        aria-valuetext= " ${challenge.rating} out of 5"
+        
+      >
+        <li class="rating-star active"></li>
+        <li class="rating-star active"></li>
+        <li class="rating-star active"></li>
+        <li class="rating-star active"></li>
+        <li class="rating-star"></li>
+      </ul>
+      <small class="challenge-meta">${challenge.minParticipants}- ${challenge.maxParticipants} participants</small>
+      <p class="challenge-description">
+       ${challenge.description}
+      </p>
+      <button class="button primary modal-open">Book this room</button>
+    </li>
 
-     `;
+    `;
```

Comment:
This way of DOM manipulation is easier to author than using the `createElement()` and `append()` methods of `document`, but it's fundamentally the same thing as first clearing a sub-tree of the DOM, and then replacing all of it's contents using those two same methods.

I know it has been confused as similar to React before, so I just want to point out that this is not doing what React does, even though syntactically some may find it similar.